app/src/features/surveys/view/SurveyMap.tsx

```diff
@@ -41,15 +82,71 @@ const SurveyMap = (props: ISurveyMapProps) => {
           <FullScreenScrollingEventHandler bounds={bounds} scrollWheelZoom={false} />
           <SetMapBounds bounds={bounds} />
 
-          {props.mapPoints?.map((nonEditableGeo: INonEditableGeometries, index: number) => {
-            const key = nonEditableGeo.feature.id ?? index;
+          {props.mapPoints?.map((mapPoint: ISurveyMapPoint, index: number) => {
+            const { key } = mapPoint;
```

Comment:
probably subjective: But I'd lean towards either de-referencing all values from mapPoint or none of them. Rather than having a mixture.

Reply:
(I tend to like de-referencing none of the values if the code is more than 1 page long, just because its easier to know what prop the values came from when reading farther than the page, but that is definitely just a personal preference)

Reply:
Agreed! I'm definitely a bit more cautious about dereferencing for this reason

Reply:
I think I did it so I could cash in on the syntactic sugar :P
```
return {
  key,
  ...
```